### ISSUE TITLE
Redirect to login when fetching profile info leads to 429 error

### ISSUE THREAD

**Issue by ekalin** [instaloader/instaloader#1882]
**Describe the bug**
For whatever reason, IG is relatively nice and did not temporarily block the account, it is just asking for a new login. Moreover, it let the initial login request succeed, but once one asks for profile data, just then it asks for a login.

This leads to messages such as:

    JSON Query to api/v1/users/web_profile_info/?username=profile: Redirected to login [retrying; skip with ^C]

Unfortunately, it seems that it just retries the query, leading to a 429 error:
```
Number of requests within last 10/11/20/22/30/60 minutes grouped by type:
   d6f4427fbe92d846298cf93df0b937d3:    1    1    1    1    1    1
 *                           iphone:   19   19   19   19   19   19
Instagram responded with HTTP error "429 - Too Many Requests". Please
do not run multiple instances of Instaloader in parallel or within
short sequence. Also, do not use any Instagram App while Instaloader
is running.
The request will be retried in 29 minutes, at 01:45.
```

Eventually, for whatever reason, it got past this requests, and was able to attempt the actual download, which failed with a 403 ("Please log back in") and caused the program to abort (because I set `--abort-on`).

**To Reproduce**
Unfortunately this is very IG dependent. It has happened twice in the last couple days, but who knows what triggers this specific IG behavior?

**Expected behavior**
If it's a simple request to login, then (if login was requested in the command line) instaloader should attempt to login again. If login succeeds, then it should attempt to continue what it was doing before. If it fails, the program should abort.

If login was not requested, then it should abort. I believe this is what happens in this case.

**Instaloader version**
4.9.5

**Additional context**
I'd be willing to submit a PR to fix this, but I'm not sure how to best solve this.

**Comment by aandergr** [comment]
Currently this login redirect, if already logged in, is handled exactly like a 429 response, i.e. with a special logic to wait and then repeat the request. This current behavior is probably not very useful, though, because the login redirect occurs after the session has been invalidated by Instagram, and so request retries following a login redirect will not succeed for the moment. 

However, Instaloader cannot re-login on its own, as this is always an interactive process (which does not even work very well at the moment, but that is another issue).

However, we could abort the download, trigger the resume logic (i.e. write a resume file for the targets that support it, and use it to resume the loop later at the same place), and write a message to the user instructing to recreate the session, wait half an hour (since the rate controller is not persisted between invocations), and restart Instaloader. This would probably be the best way to handle these login redirects while Instaloader is logged in.

**Comment by ekalin** [comment]
There have been cases where the account was logged out but the challenge was not required; simply restarting the program (causing a login) succeeded. But these seem to be an exception, most of the time it will require the challenge.

So I agree that aborting the program is a good solution.

**Comment by ekalin** [comment]
Aborting the download was quite simple, I just basically changed the exception type.

However, after aborting early (instead of retrying the request several times), in the next run the login worked (there was no need for the challenge), so maybe the repeated request were influencing IG's "suspicious behavior" detection.

So I guess it would still be worth to attempt login. But this does seem more complicated.
